src/tests/unit/base/test_queries.py

```diff
@@ -70,6 +73,77 @@ def _get_results(url, **kwargs):
     assert clonedSimple._query == simpleQuery._query
 
 
+def test_select_calls_select_class_instance():
+    """Test if the `select` method calls the `select_class_instance` function."""
+    with patch("cbc_sdk.connection.select_class_instance") as fn:
+        api = CBCloudAPI(url="https://example.com", token="ABCD/1234", org_key="WNEX", ssl_verify=True)
+        api.select("a")
+        fn.assert_called()
+
+
+def test_raise_ModelNotFound():
+    """Test ModelNotFound exception when a class isn't found."""
+    with pytest.raises(ModelNotFound):
+        api = CBCloudAPI(url="https://example.com", token="ABCD/1234", org_key="WNEX", ssl_verify=True)
+        api.select("NON_EXISTENT")
+
+
+@pytest.mark.parametrize(
+    "klass_name, query_expected", [
+        # Audit and Remediation
+        ("DeviceSummary", "ResultQuery"),
+        ("DeviceSummaryFacet", "FacetQuery"),
+        ("Result", "ResultQuery"),
+        ("ResultFacet", "FacetQuery"),
+        ("Run", "RunQuery"),
+        ("RunHistory", "RunHistoryQuery"),
+        ("Template", "RunQuery"),
+        ("TemplateHistory", "TemplateHistoryQuery"),
+
+        # Endpoint Standard
+        ("Recommendation", "RecommendationQuery"),
+        ("EnrichedEvent", "EnrichedEventQuery"),
+        ("EnrichedEventFacet", "FacetQuery"),
+        ("Policy", "Query"),
+        ("USBDevice", "USBDeviceQuery"),
+        ("USBDeviceApproval", "USBDeviceApprovalQuery"),
+        ("USBDeviceBlock", "USBDeviceBlockQuery"),
+
+        # Enterprise EDR
+        ("Feed", "FeedQuery"),
+        ("Report", "ReportQuery"),
+        ("Watchlist", "WatchlistQuery"),
+
+        # Platform
+        ("BaseAlert", "BaseAlertSearchQuery"),
+        ("CBAnalyticsAlert", "CBAnalyticsAlertSearchQuery"),
+        ("DeviceControlAlert", "DeviceControlAlertSearchQuery"),
+        ("WatchlistAlert", "WatchlistAlertSearchQuery"),
+        ("Device", "DeviceSearchQuery"),
+        ("Event", "EventQuery"),
```

Comment:
Using the following code does not reference the correct Event model
```
cb.select('Event', '7DESJ9GN-0040bdb5-000009e0-00000000-1d82fe3b49a5b32')
```

instead it matches to the following
```
<cbc_sdk.endpoint_standard.base.Event: id 7DESJ9GN-0040bdb5-000009e0-00000000-1d82fe3b49a5b32> @ https://defense.conferdeploy.net
```

Was expecting this instead
```
<cbc_sdk.platform.events.Event: id 7DESJ9GN-0040bdb5-000009e0-00000000-1d82fe3b49a5b32> @ https://defense.conferdeploy.net
```